test/unit/services/awsAdapter_spec.js

```diff
@@ -148,4 +156,166 @@ describe("The AwsAdapter class ", function () {
+			.then(function (response) {
+				result = response;
+			});
+		});
+
+		after(function () {
+			describeInstancesStub.restore();
+		});
+
+		it("gets properties of the given instance", function () {
+			expect(result).to.be.instanceof(Array);
+			expect(result[ 0 ].PublicIpAddress).to.be.equal("127.0.0.1");
+		});
+	});
+
+	describe("gets properties of invalid instance", function () {
+		var describeInstancesStub;
+		var result;
+		before(function () {
+			var awsAdapter = new AwsAdapter(ec2);
+			describeInstancesStub = Sinon.stub(ec2, "describeInstancesAsync", function () {
+				return Bluebird.reject(new Error("Instance not Found"));
+			});
+			return awsAdapter.describeInstance(VALID_EC2_INSTANCE)
+			.catch(function (error) {
+				result = error;
+			});
+		});
+
+		after(function () {
+			describeInstancesStub.restore();
+		});
+
+		it("fails", function () {
+			expect(result).to.be.an.instanceof(Error);
+			expect(result.message).to.equal("Instance not Found");
+		});
+	});
+
+	describe("gets the IP Address of the instance", function () {
+		var result;
+		var describeInstancesStub;
+
+		before(function () {
+			var awsAdapter = new AwsAdapter(ec2);
+			describeInstancesStub = Sinon.stub(ec2, "describeInstancesAsync", function () {
+				var data = { Reservations : [ { Instances : [ { PublicIpAddress : "127.0.0.1" } ] } ] };
+				return Bluebird.resolve(data);
+			});
+
+			return awsAdapter.getPublicIPAddress(VALID_EC2_INSTANCE)
+			.then(function (response) {
+				result = response;
+			})
+			.catch(function (error) {
+				result = error;
+			});
+		});
+
+		after(function () {
+			describeInstancesStub.restore();
+		});
+
+		it("gets a valid IP address", function () {
+			expect(describeInstancesStub.called).to.be.true;
+			expect(result).to.equal(VALID_IP_ADDRESS);
+		});
+	});
+
+	describe("begins Polling", function () {
+
+		var sshAdapter = new SshAdapter(ec2);
+		var instances  = new InstanceAdapter();
+		var options    = {};
+		var serverLog  = function () {
+			//This is an empty block which mocks server.log function
+		};
+		options.sshAdapter = sshAdapter;
+
+		var awsAdapter = new AwsAdapter(ec2, options);
+
+		describe("and gets IP address of instance", function () {
+			var startSshPollingStub;
+			var getPublicIPAddressStub;
+			var instanceProp;
+
+			before(function () {
+				startSshPollingStub = Sinon.stub(sshAdapter,"startSshPolling", function () {
+					return Bluebird.resolve();
+				});
+
+				getPublicIPAddressStub = Sinon.stub(awsAdapter,"getPublicIPAddress", function () {
+					return Bluebird.resolve(VALID_IP_ADDRESS);
+				});
+
+				return instances.createInstance("ami-d05e75b8","t2.micro")
+				.then(function (data) {
+					instanceProp  = JSON.parse(JSON.stringify(data));
+				});
+			});
+
+			after(function () {
+				startSshPollingStub.restore();
+				getPublicIPAddressStub.restore();
+			});
+
+			it("updates the state and uri of the instance", function () {
```

Comment:
This seems to only check one side of this scenario. I believe this test will pass even if `updateInstance()` is never called.